stubs/openpyxl/openpyxl/styles/colors.pyi

```diff
@@ -66,11 +66,11 @@ class ColorDescriptor(Typed[Color, _N]):
         self: ColorDescriptor[Literal[False]], name: str | None = None, *, allow_none: Literal[False] = False
     ) -> None: ...
     @overload
-    def __set__(self: ColorDescriptor[_N], instance: Serialisable | Strict, value: str) -> None: ...
-    @overload
     def __set__(self: ColorDescriptor[Literal[True]], instance: Serialisable | Strict, value: Color | None) -> None: ...
     @overload
     def __set__(self: ColorDescriptor[Literal[False]], instance: Serialisable | Strict, value: Color) -> None: ...
+    @overload
+    def __set__(self: ColorDescriptor[_N], instance: Serialisable | Strict, value: str) -> None: ...
```

Comment:
This looks like the correct implementation, maybe I mixed something up or a bad copy paste when I did https://github.com/python/typeshed/pull/10882

```suggestion
    def __set__(self: ColorDescriptor[Literal[True]], instance: Serialisable | Strict, value: str | Color | None) -> None: ...
    @overload
    def __set__(self: ColorDescriptor[Literal[False]], instance: Serialisable | Strict, value: str | Color) -> None: ...
```

This also matches the usages of `ColorDescriptor` in the stubs


> I'm not sure I understand why this class is generic

@JelleZijlstra That generic is assigned to the `allow_none` property. It defines whether you can use `None` in the Descriptor setter.